pkg/fileutil/fileutil.go

```diff
@@ -238,3 +238,14 @@ func IsExecutable(path string) bool {
 	}
 	return false
 }
+
+// IsDeviceNode checks if the given path points to a block or char device.
+// It doesn't follow symlinks.
+func IsDeviceNode(path string) bool {
+	d, err := os.Lstat(path)
+	if err == nil {
+		m := d.Mode()
+		return m&os.ModeDevice == os.ModeDevice || m&os.ModeCharDevice == os.ModeCharDevice
```

Comment:
Testing `os.ModeDevice` is enough because in Golang, ModeDevice and ModeCharDevice are not exclusive, as opposed to syscall.S_IFCHR and syscall.S_IFBLK...

See:
https://golang.org/pkg/os/#FileMode
```
ModeDevice             // D: device file
ModeCharDevice         // c: Unix character device, when ModeDevice is set
```
And:
https://golang.org/src/os/stat_linux.go#L18